src/transformers/models/llama/modeling_llama.py

```diff
@@ -439,7 +444,12 @@ def forward(
 
         # upcast attention to fp32
         attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(query_states.dtype)
-        attn_output = torch.matmul(attn_weights, value_states)
+        # attn_weights Batch Num_head Seq Kv_seq
+        # value_states Batch Num_head Seq Head_dim
+        # attn_output = torch.matmul(attn_weights, value_states)
+        assert attn_weights.shape == torch.Size((bsz, self.num_heads, q_len, kv_seq_len)), f'incorrect atten_weight shape: {attn_weights.shape}'
+        assert value_states.shape == torch.Size((bsz, self.num_heads, q_len, self.head_dim)), f'incorrect value_states shape: {value_states.shape}'
+        attn_output = torch.einsum('bnsk,bnsh->bnsh', attn_weights, value_states)
```

Comment:
Why there is a transpose here as well since the original code doesn't?

Reply:
This should be `bnsk,bnkh->bnsh`